lightning/src/util/ser.rs

```diff
@@ -34,65 +34,71 @@ use util::byte_utils::{be64_to_array, be48_to_array, be32_to_array, be16_to_arra
 /// serialization buffer size
 pub const MAX_BUF_SIZE: usize = 64 * 1024;
 
-/// A trait that is similar to std::io::Write but has one extra function which can be used to size
-/// buffers being written into.
-/// An impl is provided for any type that also impls std::io::Write which simply ignores size
-/// hints.
+/// A struct that implements a serialization buffer.  A size hint can be given and the length
+/// of the result can be retrieved.
 ///
-/// (C-not exported) as we only export serialization to/from byte arrays instead
-pub trait Writer {
+/// std::io::Write is implemented for serializing Bitcoin objects with impl_consensus_ser! below.
+pub struct Writer(pub Vec<u8>);
+
+impl Writer {
+	/// Construct an empty Writer
+	pub fn new() -> Self {
+		Writer(Vec::new())
+	}
+
 	/// Writes the given buf out. See std::io::Write::write_all for more
-	fn write_all(&mut self, buf: &[u8]) -> Result<(), ::std::io::Error>;
+	pub fn write_all(&mut self, buf: &[u8]) -> Result<(), ::std::io::Error> {
```

Comment:
`#[inline]` seems desirable given https://stackoverflow.com/questions/37639276/when-should-inline-be-used-in-rust, so done that.  But it seems strange to force the compiler with `always` rather than give a hint - I doubt the compiler will fail to take advantage unless there's a good reason given the rest of the caller's code.